From 7a840621181bc1f6301149da646c8672d8a0afbb Mon Sep 17 00:00:00 2001
From: Vaibhav Hiremath <hvaibhav@ti.com>
Date: Fri, 16 Jan 2009 13:24:41 +0530
Subject: [PATCH 180/426] old FBDEV made working

With new DSS2 and FBDEV driver the old FBDEV driver was
broken, fixed the board-omap3evm.c file for the same
---
 arch/arm/mach-omap2/board-omap3evm.c |   52 ++++++++++++++++++++++++----------
 arch/arm/plat-omap/dss/Kconfig       |   10 +++---
 drivers/video/omap/Kconfig           |    1 +
 drivers/video/omap2/Kconfig          |    8 +++--
 4 files changed, 48 insertions(+), 23 deletions(-)

diff --git a/arch/arm/mach-omap2/board-omap3evm.c b/arch/arm/mach-omap2/board-omap3evm.c
index c8c3973..48985bf 100644
--- a/arch/arm/mach-omap2/board-omap3evm.c
+++ b/arch/arm/mach-omap2/board-omap3evm.c
@@ -62,20 +62,6 @@ static struct platform_device omap3evm_smc911x_device = {
 	.resource	= &omap3evm_smc911x_resources [0],
 };
 
-#ifdef CONFIG_FB_OMAP2
-static struct resource omap3evm_vout_resource[3 - CONFIG_FB_OMAP2_NUM_FBS] = {
-};
-#else
-static struct resource omap3evm_vout_resource[2] = {
-};
-#endif
-
-static struct platform_device omap3evm_vout_device = {
-	.name			= "omap_vout",
-	.num_resources	= ARRAY_SIZE(omap3evm_vout_resource),
-	.resource 		= &omap3evm_vout_resource[0],
-	.id		= -1,
-};
 static inline void __init omap3evm_init_smc911x(void)
 {
 	int eth_cs;
@@ -173,6 +159,24 @@ static int __init omap3_evm_i2c_init(void)
 	omap_register_i2c_bus(3, 400, NULL, 0);
 	return 0;
 }
+/*
+ * For new frame buffer driver based on DSS2 library
+ */
+#ifdef CONFIG_OMAP2_DSS
+
+#ifdef CONFIG_FB_OMAP2
+static struct resource omap3evm_vout_resource[3 - CONFIG_FB_OMAP2_NUM_FBS] = {
+};
+#else /* CONFIG_FB_OMAP2 */
+static struct resource omap3evm_vout_resource[2] = {
+};
+#endif /* CONFIG_FB_OMAP2 */
+static struct platform_device omap3evm_vout_device = {
+	.name			= "omap_vout",
+	.num_resources	= ARRAY_SIZE(omap3evm_vout_resource),
+	.resource 		= &omap3evm_vout_resource[0],
+	.id		= -1,
+};
 
 #define LCD_PANEL_LR		2
 #define LCD_PANEL_UD		3
@@ -368,6 +372,15 @@ static struct platform_device omap3_evm_dss_device = {
 		.platform_data = &omap3_evm_dss_data,
 	},
 };
+#else /* CONFIG_OMAP2_DSS */
+static struct platform_device omap3_evm_lcd_device = {
+	.name		= "omap3evm_lcd",
+	.id		= -1,
+};
+static struct omap_lcd_config omap3_evm_lcd_config __initdata = {
+	.ctrl_name	= "internal",
+};
+#endif /* CONFIG_OMAP2_DSS */
 
 static void ads7846_dev_init(void)
 {
@@ -425,12 +438,19 @@ static void __init omap3_evm_init_irq(void)
 
 static struct omap_board_config_kernel omap3_evm_config[] __initdata = {
 	{ OMAP_TAG_UART,	&omap3_evm_uart_config },
+#ifndef CONFIG_OMAP2_DSS
+	{ OMAP_TAG_LCD,		&omap3_evm_lcd_config },
+#endif
 };
 
 static struct platform_device *omap3_evm_devices[] __initdata = {
+#ifdef CONFIG_OMAP2_DSS
 	&omap3_evm_dss_device,
-	&omap3evm_smc911x_device,
 	&omap3evm_vout_device,
+#else /* CONFIG_OMAP2_DSS */
+	&omap3_evm_lcd_device,
+#endif /* CONFIG_OMAP2_DSS */
+	&omap3evm_smc911x_device,
 
 };
 
@@ -461,7 +481,9 @@ static void __init omap3_evm_init(void)
 	usb_ehci_init();
 	omap3evm_flash_init();
 	ads7846_dev_init();
+#ifdef CONFIG_OMAP2_DSS
 	omap3_evm_display_init();
+#endif /* CONFIG_OMAP2_DSS */
 }
 
 static void __init omap3_evm_map_io(void)
diff --git a/arch/arm/plat-omap/dss/Kconfig b/arch/arm/plat-omap/dss/Kconfig
index f0b1f1c..7aad779 100644
--- a/arch/arm/plat-omap/dss/Kconfig
+++ b/arch/arm/plat-omap/dss/Kconfig
@@ -8,14 +8,14 @@ if OMAP2_DSS
 
 config OMAP2_DSS_DEBUG_SUPPORT
         bool "Debug support"
-	default y
+	default n
 	help
 	  This enables debug messages. You need to enable printing
 	  with 'debug' module parameter.
 
 config OMAP2_DSS_RFBI
 	bool "RFBI support"
-        default y
+        default n
 
 config OMAP2_DSS_VENC
 	bool "VENC support"
@@ -25,11 +25,11 @@ if ARCH_OMAP3
 
 config OMAP2_DSS_SDI
 	bool "SDI support"
-        default y
+        default n
 
 config OMAP2_DSS_DSI
 	bool "DSI support"
-        default y
+        default n
 
 endif
 
@@ -53,7 +53,7 @@ config OMAP2_DSS_FAKE_VSYNC
 config OMAP2_DSS_MIN_FCK_PER_PCK
 	int "Minimum FCK/PCK ratio (for scaling)"
 	range 0 32
-	default 0
+	default 4
 	help
 	  This can be used to adjust the minimum FCK/PCK ratio.
 
diff --git a/drivers/video/omap/Kconfig b/drivers/video/omap/Kconfig
index 541fab3..463e907 100644
--- a/drivers/video/omap/Kconfig
+++ b/drivers/video/omap/Kconfig
@@ -5,6 +5,7 @@ config FB_OMAP
 	select FB_CFB_FILLRECT
 	select FB_CFB_COPYAREA
 	select FB_CFB_IMAGEBLIT
+	default y if (OMAP2_DSS = "n")
 	help
           Frame buffer driver for OMAP based boards.
 
diff --git a/drivers/video/omap2/Kconfig b/drivers/video/omap2/Kconfig
index 4e9211e..ddb1338 100644
--- a/drivers/video/omap2/Kconfig
+++ b/drivers/video/omap2/Kconfig
@@ -5,12 +5,13 @@ config FB_OMAP2
         select FB_CFB_FILLRECT
         select FB_CFB_COPYAREA
         select FB_CFB_IMAGEBLIT
+	default OMAP2_DSS if (FB_OMAP = "n")
         help
           Frame buffer driver for OMAP2/3 based boards.
 
 config FB_OMAP2_DEBUG
 	bool "Debug support for OMAP2/3 FB"
-	default y
+	default n
 	depends on FB_OMAP2
 	help
 	  Support for debug output. You have to enable the actual printing
@@ -27,7 +28,7 @@ config FB_OMAP2_FORCE_AUTO_UPDATE
 
 config FB_OMAP2_NUM_FBS
 	int "Number of framebuffers"
-	range 1 10
+	range 1 3
 	default 3
 	depends on FB_OMAP2
 	help
@@ -39,13 +40,14 @@ menu "OMAP2/3 Display Device Drivers"
 
 config PANEL_GENERIC
         tristate "Generic Panel"
+	default y
         help
 	  Generic panel driver.
 	  Used for DVI output for Beagle and OMAP3 SDP.
 
 config PANEL_SHARP_LS037V7DW01
         tristate "Sharp LS037V7DW01 LCD Panel"
-        depends on OMAP2_DSS
+	default y
         help
           LCD Panel used in TI's SDP3430 and EVM boards
 
-- 
1.6.2.4

